File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func deepestLeavesSum(root *TreeNode) int {
	var deepestLevel, sum int
	var dfs func(*TreeNode, int)
	dfs = func(node *TreeNode, level int) {
		
        if node == nil {
            return
        }
        
        if level == deepestLevel {
			sum += node.Val
		}
        
        if level > deepestLevel {
			deepestLevel = level
			sum = node.Val
		}

		
		dfs(node.Left, level+1)
		dfs(node.Right, level +1)
	}

	dfs(root, 0)

	return sum
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        int answer = 0;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while (queue.size() > 0) {
            answer = 0;
            int queueSize = queue.size();
            for (int i = 0; i < queueSize; i++) {
                TreeNode node = queue.poll();
                answer += node.val;

                if (node.left != null) {
                    queue.offer(node.left);
                }

                if (node.right != null) {
                    queue.offer(node.right);
                }
            }
        }

        return answer;
    }
}